File: EK_Static/js/newsapi.js
/*
 * Created by Leo N. on 2017-08-03.
 */


$(document).ready(function () {
    NewsApi();
});


// Preparing the HTML
function NewsApi() {

    // Variables
    var apikey = 'f355c018b1474aef93c183aebcb3b845'; // get an API Key from 'https://newsapi.org'
    var req = new XMLHttpRequest();

    var news = document.getElementById('news');
    var index = 0;


    req.open('GET', 'https://newsapi.org/v2/top-headlines?sources=bbc-news,associated-press,abc-news,cbc-news,engadget,the-economist,the-guardian-au&apiKey=' + apikey);
    // What happens when the request loads...
    req.onload = function () {
        // console.log(JSON.parse(req.responseText));
        var parsedResponse = JSON.parse(req.responseText);
        var articles = parsedResponse.articles;
        var length = parsedResponse.articles.length;
        var main_source = parsedResponse.source;

        console.log("Number of articles: " + length);
        var article = "";
        var featuredContent = "";
        var featuredAuthor = "";
        var featuredSource = "";
        var link = "";
        var random = Math.round(Math.random() * (length - 1) + 1);

        if (random > length) {
            // console.log(random);
            random = Math.round(Math.random() * (length - 1) - 1);
            if (articles[random].author === "") {
                random = Math.round(Math.random() * (length - 1) / 2);
            }
        }

        for (index; index < length; index++) {
            article += '<div class="p-top-60 p-bottom-40">';
                 article += '<div class="blog-content">';
                    article += '<div class="col-md-9 mr-auto">';
                        article += '<p><span>'+ articles[index].source.id +'</span></p>';
                        article += '<a href="'+ articles[index].url +'" target="_blank">';
                            article += '<h1 class="p-bottom-10">'+ articles[index].title +'</h1>';
                        article += '</a>';
                        article += '<img class="img-fluid" src="'+ articles[index].urlToImage +'">';
                        article += '<div class="p-top-10">'+ articles[index].description +'</div>';
                    article += '</div>';
                 article += '</div>';
            article += '</div>';
        }

        news.innerHTML = article ;
    };
    req.send();

    var requestAPI = setInterval(NewsApi, 120000);
}

File: EKSite/newsapi.py
"""

 newsapi.py
 By Leo N.
 An implementation of the News API.
 Requirements: Python 3+, Internet, News API Key.

"""

# Importing Python web modules
import random, json, urllib.parse, requests

# Developer API key
key = "f355c018b1474aef93c183aebcb3b845"


# Articles array
NEWS = []
ARTICLES = []
TITLES = []

# API Endpoint: Querying News
api = 'https://newsapi.org/v2/top-headlines?sources=bbc-news,associated-press,abc-news,cbc-news,engadget,the-economist,the-guardian-au&apiKey='+key


def NewsObjects():
    data = requests.get(
        api,
        verify = True,
    ).json()

    size = len(data['articles'])

    for i in range(size):
        ARTICLES.insert(i, data['articles'][i])



    count = 0

    for news in ARTICLES:
        # print(news['title'])
        TITLES.insert(i, news['title'])
        if news['title'] not in TITLES:
            NEWS.insert(count, news)
            # print(NEWS['articles'][count]['title'])
        count = count + 1

    return ARTICLES



# ------------------------------- #

def main():
    news = NewsObjects()

# ------------------------------- #

if __name__ == '__main__':
    main()
